pkg/parser/jsonschema_test.go

```diff
@@ -72,3 +73,265 @@ test:
+			yaml: `
+version: 2.1
+jobs:
+  my-lock-job:
+    type: lock
+    key: my-key
+    some_other_property: allowed
+`,
+			expectError: false,
+		},
+		{
+			name: "lock type - missing key (should error)",
+			yaml: `
+version: 2.1
+jobs:
+  my-lock-job:
+    type: lock
+`,
+			expectError:         true,
+			expectErrorContains: "key",
+		},
+
+		// Unlock type tests
+		{
+			name: "unlock type - valid with key",
+			yaml: `
+version: 2.1
+jobs:
+  my-unlock-job:
+    type: unlock
+    key: my-lock-key
+`,
+			expectError: false,
+		},
+		{
+			name: "unlock type - valid with additional properties",
+			yaml: `
+version: 2.1
+jobs:
+  my-unlock-job:
+    type: unlock
+    key: my-key
+    some_other_property: allowed
+`,
+			expectError: false,
+		},
+		{
+			name: "unlock type - missing key (should error)",
+			yaml: `
+version: 2.1
+jobs:
+  my-unlock-job:
+    type: unlock
+`,
+			expectError:         true,
+			expectErrorContains: "key",
+		},
+
+		// Approval type tests
+		{
+			name: "approval type - valid minimal",
+			yaml: `
+version: 2.1
+jobs:
+  my-approval-job:
+    type: approval
+`,
+			expectError: false,
+		},
+		{
+			name: "approval type - valid with steps (ignored)",
+			yaml: `
+version: 2.1
+jobs:
+  my-approval-job:
+    type: approval
+    steps:
+      - run: echo "This will be ignored"
+`,
+			expectError: false,
+		},
+
+		// No-op type tests
+		{
+			name: "no-op type - valid minimal",
+			yaml: `
+version: 2.1
+jobs:
+  my-noop-job:
+    type: no-op
+`,
+			expectError: false,
+		},
+		{
+			name: "no-op type - valid with steps (ignored)",
+			yaml: `
+version: 2.1
+jobs:
+  my-noop-job:
+    type: no-op
+    steps:
+      - run: echo "This will be ignored"
+`,
+			expectError: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			context := testHelpers.GetDefaultLsContext()
+			yamlDocument, _ := ParseFromContent([]byte(tc.yaml), context, uri.File(""), protocol.Position{})
+
+			if tc.expectError {
+				// For error cases, also run JSON schema validation
+				validator := JSONSchemaValidator{
+					Doc: yamlDocument,
+				}
+				// Use relative path to schema.json from pkg/parser
+				schemaPath := "../../schema.json"
+				err := validator.LoadJsonSchema(schemaPath)
+				if err != nil {
+					t.Logf("Warning: Could not load schema: %v", err)
+					t.SkipNow()
+				}
+
+				diagnostics := validator.ValidateWithJSONSchema(yamlDocument.RootNode, yamlDocument.Content)
+
+				// Log all diagnostics for debugging
+				if len(diagnostics) > 0 {
+					t.Logf("Found %d diagnostic(s):", len(diagnostics))
+					for _, d := range diagnostics {
+						t.Logf("  - %s", d.Message)
+					}
+				} else {
+					t.Logf("No diagnostics found")
+				}
+
+				// TODO: These tests are currently failing because gojsonschema may not
+				// fully support JSON Schema draft-06 if/then/else conditionals.
+				// Uncomment the assertions below once the schema validation is fixed.
```

Comment:
I don't see any commented-out assertions below here, so maybe this comment itself is outdated?

Reply:
Oh! Good catch, I managed to fix it by bumping the schema version but forgot to remove this comment. Thanks!